dashboard/lib/pd/foorm/rollup_helper.rb

```diff
@@ -71,23 +70,26 @@ def self.get_rollup_question_data(parsed_forms, question_details)
       questions
     end
 
-    # Validate question is valid for rolling up. Validations are:
-    #   - question is the same across all forms it was found in. Same is defined by
-    #   having the same type and same choices/rows/columns
+    # Validate question is valid for rolling up and return question if valid. Validations are:
+    #   - question is the same across all forms it was found in.
+    #    "Same" is defined as having the same type and same choices/columns.
+    #    If there are differences in rows, merge the rows together.
+    #    If there are any conflicts, take the text for the first row with a given id.
     #   - question answers can be parsed as integers
-    def self.validate_question(question, question_data, parsed_forms, form_type)
-      rows = nil
+    def self.validate_and_get_question(question, question_data, parsed_forms, form_type)
       columns = nil
+      question_to_return = nil
       question_data[:form_keys].each do |form_key|
         parsed_question = parsed_forms[form_type][form_key][question]
         return false unless parsed_question && parsed_question[:type] = question_data[:type]
+        question_to_return ||= parsed_question
```

Comment:
More for my understanding of this operator than anything...`question_to_return` is definitely nil, so is this the same as `question_to_return = parsed question`?

Reply:
on the first iteration through the loop it's the same, but we only want to overwrite question_to_return if it's nil--otherwise we keep the first question, possibly updating the rows as we go.

Reply:
d'oh. thanks.